### PR TITLE
zplug: init at 2.4.2.

### PR DESCRIPTION
**Pull request by kayhide**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/shells/zsh/zplug/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  version = "2.3.4";
+  name = "zplug-${version}";
```
Comment (mmahut) [inline pkgs/shells/zsh/zplug/default.nix]: Please consider using ``pname``.

Comment (mmahut) [inline pkgs/shells/zsh/zplug/default.nix]:
```suggestion
  pname = "zplug";
```

Please consider using ``pname``.

File: pkgs/shells/zsh/zplug/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  phases = "installPhase";
+
+  installPhase = ''
+  mkdir -p $out/opt
+  cp -r $src $out/opt/zplug
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A next-generation plugin manager for zsh.";
```
Comment (mmahut) [inline pkgs/shells/zsh/zplug/default.nix]:
```suggestion
    description = "A next-generation plugin manager for zsh";
```

Please remove dot at the end of the description.

File: pkgs/shells/zsh/zplug/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "zplug";
+    repo = "zplug";
+    rev = version;
+    sha256 = "1wcvk68fwyfba3kcin6fl0ifwf9mr8r914kq132nk96ga625j8b1";
+  };
+
+  phases = "installPhase";
+
+  installPhase = ''
+  mkdir -p $out/opt
```
Comment (mmahut) [inline pkgs/shells/zsh/zplug/default.nix]: If possible indent this block.

File: pkgs/shells/zsh/zplug/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  version = "2.3.4";
```
Comment (mmahut) [inline pkgs/shells/zsh/zplug/default.nix]: The latest version is ``2.4.2``.

File: pkgs/shells/zsh/zplug/default.nix (hunk 1 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: Okay, I will do it.

File: pkgs/shells/zsh/zplug/default.nix (hunk 2 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: Got it.

File: pkgs/shells/zsh/zplug/default.nix (hunk 3 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: I will do it.

File: pkgs/shells/zsh/zplug/default.nix (hunk 4 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: Oh, that is true, I will update it.

**Comment by kayhide** [comment]
@mmahut Thanks for your review. I updated all items you pointed.
Please take a look.

**Comment by mmahut** [comment]
@GrahamcOfBorg build zplug

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -2669,6 +2669,11 @@
     email = "info+nix@chmist.com";
     name = "karolchmist";
   };
+  kayhide = {
```
Comment (mmahut) [inline maintainers/maintainer-list.nix]: Do you mind having this as a separate commit?

Comment (kayhide) [reply maintainers/maintainer-list.nix]: I will do it!

Comment (kayhide) [reply maintainers/maintainer-list.nix]: Done!

**Comment by mmahut** [comment]
@GrahamcOfBorg build zplug

**Comment by kayhide** [comment]
As commented at https://github.com/NixOS/nixpkgs/pull/66629/files#r319707103 , I updated maintainer list and added github id.

**Comment by mmahut** [comment]
@GrahamcOfBorg build zplug

File: pkgs/shells/zsh/zplug/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "zplug";
+
+  src = fetchFromGitHub {
+    owner = "zplug";
+    repo = "zplug";
+    rev = version;
+    sha256 = "0hci1pbs3k5icwfyfw5pzcgigbh9vavprxxvakg1xm19n8zb61b3";
+  };
+
+  phases = "installPhase";
```
Comment (aanderse) [inline pkgs/shells/zsh/zplug/default.nix]: Please avoid use of `phases` if possible. https://github.com/NixOS/nixpkgs/issues/28910

File: pkgs/shells/zsh/zplug/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  installPhase = ''
```
Comment (aanderse) [inline pkgs/shells/zsh/zplug/default.nix]: Please include a `runHook preInstall` and `runHook postInstall` as the first and last line of the `installPhase` if you don't mind.

File: pkgs/shells/zsh/zplug/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    description = "A next-generation plugin manager for zsh";
+    longDescription = ''
+      ZPlug is a next-generation plugin manager for zsh.
+
+      To use zplug, add the following to your .zshrc:
+
+        export ZPLUG_HOME=$HOME/.zplug
+        source $(nix-env -q --out-path zplug | cut -d ' ' -f 3)/opt/zplug/init.zsh
+    '';
+    homepage = https://zplug.sh/;
```
Comment (aanderse) [inline pkgs/shells/zsh/zplug/default.nix]: Please quote.

**Review by aanderse — CHANGES_REQUESTED** [review]
I've left a few comments I hope you find helpful.

File: pkgs/shells/zsh/zplug/default.nix (hunk 8 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: Got it!

File: pkgs/shells/zsh/zplug/default.nix (hunk 7 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: Got it!

File: pkgs/shells/zsh/zplug/default.nix (hunk 6 above)
Comment (kayhide) [reply pkgs/shells/zsh/zplug/default.nix]: I removed `phases` and added the following instead.

```
  configurePhase = ''

  '';

  buildPhase = ''

  '';
```

Is this what we want?

**Comment by kayhide** [comment]
Same PR was open at #83856 and introduced at 71a768a6e814b0198ec0ab82735d27d44f5a8c31

Closing
